packages/regl-worldview/src/Worldview.js

```diff
@@ -70,12 +68,32 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
 
   constructor(props: BaseProps) {
     super(props);
-    const { width, height, top, left, backgroundColor } = props;
+    const { width, height, top, left, backgroundColor, onCameraStateChange, cameraState, defaultCameraState } = props;
+    if (onCameraStateChange) {
+      if (!cameraState) {
+        console.warn(
+          "Worldview camera state is read-only as you provided `onCameraStateChange` prop without `cameraState` prop. Turn worldview to controlled with `cameraState` and `onCameraStateChange` props, or uncontrolled with `defaultCameraState` prop."
+        );
+      }
+      if (cameraState && defaultCameraState) {
+        console.warn(
+          "You provided `cameraState`, `onCameraStateChange` and `defaultCameraState` props. `defaultCameraState` will be ignored for Worldview is controlled when both `cameraState` and `onCameraStateChange` are present."
+        );
+      }
+    } else {
+      if (cameraState) {
+        console.warn(
+          "Failed prop type: you provided `cameraState` prop without an `onCameraStateChange` handler. This will render a read-only field. If the field should be mutable use `defaultCameraState`. Otherwise set `onCameraStateChange`."
```

Comment:
I'm not sure this should be a warning, seems like a valid use case to have a fixed camera. But if you want to keep it here are some wording suggestions:
```suggestion
           "You provided `cameraState` without an `onCameraStateChange` handler. This will prevent moving the camera. If the camera should be movable, use `defaultCameraState`, otherwise set `onCameraStateChange`."
```